app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -390,6 +397,24 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
     }
 
+    private void loginDemoUser() {
+        OfflineUserRestore offlineUserRestore = CommCareApplication._().getCommCarePlatform().getDemoUserRestore();
+        if (offlineUserRestore != null) {
+            loginCczDemoUser(offlineUserRestore);
+        } else {
+            DemoUserBuilder.build(this, CommCareApplication._().getCurrentApp());
+            tryLocalLogin(DemoUserBuilder.DEMO_USERNAME, DemoUserBuilder.DEMO_PASSWORD, false,
+                    false, LoginMode.PASSWORD, true);
+        }
+    }
+
+    private void loginCczDemoUser(OfflineUserRestore offlineUserRestore) {
+        String username = offlineUserRestore.getUsername();
+        String password = offlineUserRestore.getPassword();
+        performingCczDemoUserLogin = true;

```

Comment:
I'm a bit worried about this boolean's status being a dependence inside of this class, as I don't think this class really uses member variables for this kind of state. Is there a specific reason that this flag isn't getting passed through the method instead?